boomer/src/main.rs

```diff
@@ -43,14 +43,29 @@ struct CliArgs {
     verbose: u8,
 }
 
-fn find_connector(device: &Device, connector_name: Option<&str>) -> Option<Rc<Connector>> {
-    if let Some(name) = connector_name {
-        device.connectors().find(|con| con.to_string() == name)
```

Comment:
This one can be:

```rust
device
    .connectors()
    .find(|con| con.to_string() == name)
    .ok_or(anyhow::anyhow!("No active connectors"))
```

And will handle the case where we should have had a single connector, but didn't.